docs/protocol.rst

```diff
@@ -0,0 +1,60 @@
+Tunneldigger protocol
+#####################
+
+The Tunneldigger protocol is based on L2TPv3.
+Tunneldigger implementing a custom control protocol,
+while user data is encapsulated as L2TPv3 specifies.
+This allows to use in kernel acceleration of the data path.
+
+The L2TPv3 is specified in `RFC3931 <https://tools.ietf.org/html/rfc3931>`_
+
+Control packets
+***************
+
+Tunneldigger is not following RFC3931 for control messages.
+Tunneldigger is only supporting the T bit in the first bit to
+mark a packet as control data. All other fields of RFC3931
+are ignore and have a different meaning..
+All fields are encoding as network byte order.
+
+.. packetdiag::
+
+    {
+        colwidth = 16
+        node_height = 48
+
+        0:      T
+        1-7:    0
+        8-23:   Magic 0x73A7
+        24-31:  Version
+        32-39:  Type
+        40-47:  Length
+        48-63:  Value
+    }
+
+
+* The T bit must be 1
+* Version must be 1
+* Type of the PDU
```

Comment:
What's a "PDU"?

Reply:
Tunneldigger calls these CONTROL_TYPE or msg_type; that might be more suited terminology.

Reply:
Protocol data unit https://en.wikipedia.org/wiki/Protocol_data_unit

Reply:
I see. I don't think we use this term anywhere in tunneldigger though?